integration-test/src/main/java/org/apache/iotdb/itbase/runtime/SortedClusterTestResultSet.java

```diff
@@ -0,0 +1,1218 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.itbase.runtime;
+
+import org.apache.iotdb.itbase.exception.InconsistentDataException;
+
+import java.io.InputStream;
+import java.io.Reader;
+import java.math.BigDecimal;
+import java.net.URL;
+import java.sql.Array;
+import java.sql.Blob;
+import java.sql.Clob;
+import java.sql.Date;
+import java.sql.NClob;
+import java.sql.Ref;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.RowId;
+import java.sql.SQLException;
+import java.sql.SQLWarning;
+import java.sql.SQLXML;
+import java.sql.Statement;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class SortedClusterTestResultSet implements ResultSet {
+
+  private final List<ResultSet> resultSets;
+  private final List<String> endpoints;
+  private final int queryTimeoutSeconds;
+  private final List<List<StringBuilder>> cachedResult = new ArrayList<>();
+
+  public SortedClusterTestResultSet(
+      List<Statement> statements, List<String> endpoints, String sql, int queryTimeoutSeconds)
+      throws SQLException {
+    this.queryTimeoutSeconds = queryTimeoutSeconds;
+    this.endpoints = endpoints;
+    RequestDelegate<ResultSet> delegate = createRequestDelegate();
+    for (Statement st : statements) {
+      delegate.addRequest(() -> st.executeQuery(sql));
+      cachedResult.add(new ArrayList<>());
+    }
+    resultSets = delegate.requestAll();
+  }
+
+  @Override
+  public boolean next() throws SQLException {
+    RequestDelegate<Boolean> delegate = createRequestDelegate();
+    for (ResultSet rs : resultSets) {
+      delegate.addRequest(rs::next);
+    }
+
+    boolean result = delegate.requestAllAndCompare();
+    if (result) {
+      for (List<StringBuilder> list : cachedResult) {
+        list.add(new StringBuilder());
+      }
+    }
+    return result;
+  }
+
+  @Override
+  public void close() throws SQLException {
```

Comment:
As we have enriched the logic in `close()` when using disordered result set, it implies that the developers should remember to close the dataset after usage. Is there any way to force to check the `Connection`, `Statement` and `ResultSet` closed? Here I have two ideas:
1. Use `checkstyle` to prevent unclosed resources, which will be a prefect and simple solution but I'm not sure whether `checkstyle` can really do this.
2. When closing an upper layer resource, It should check whether the direct lower layer resources have been closed. Otherwise it should throw an Exception. So we need to record the allocations of Connection in an Env, allocations of Statement in an Connection, and the allocations of ResultSet in an Statement. After an Env is destoryed, it will check whether there are any unclosed connections. If so, thrown an Exception to tell the developers. So do the Connection and Statement.